TeamNovaSoft/discord-bot
# Feat/iss 19: Configure Husky with ESLint, Prettier, and Commitlint in pre-commit

### Description

This PR adds the configuration of Husky to run ESLint, Prettier, and Commitlint in the pre-commit, ensuring that all code committed is formatted and adheres to the established commit conventions.

- Installed Husky, lint-staged, @commitlint/config-conventional and @commitlint/cli.
- Created pre-commit hook to run lint-staged.
- Created commitlint.config.js file with conventional commit configuration.
- Created commit-msg hook to validate commit messages.

Fixes:
 - #19 

### How to test it

Make changes to any file in the project. Try committing and verify that Husky runs ESLint, Prettier, and Commitlint before allowing the commit.

### Checklist
 This issue can be closed when the following tasks are complete:
 
 - [x] Husky run in pre-commit EsLint, Prettier and Conventional Commit

### Screenshots

>![image](https://github.com/user-attachments/assets/e5bef28e-1733-47f5-85bc-69b8702d7a23)

>![image](https://github.com/user-attachments/assets/a5f396ae-2add-4876-810c-a685274e5496)
